### PR TITLE
KAZOO-5697: JWT refresh, pvt_signature_secret resets

### PR DESCRIPTION
**Pull request by danielfinke**
https://forums.2600hz.com/forums/topic/9033-oauth-2-server-refresh-tokens-jwt/

Updated my previous approach based on @jamesaimonetti 's points in the above forum post.

### CODE REVIEW

**Comment by lazedo** [comment]
@danielfinke thanks for doing this. as first comment, refresh token api should be moved to cb_auth

**Comment by danielfinke** [comment]
@lazedo OK, will do that. Just for my reference, are we trying to deprecate cb_token_auth?

**Comment by lazedo** [comment]
@danielfinke no, cb_token_auth acts more like `authorization` not as an token issuer

**Comment by danielfinke** [comment]
@lazedo moved the API to cb_auth

**Comment by lazedo** [comment]
@danielfinke will review asap

File: applications/crossbar/doc/auth.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ If you feel that an account or a user secret is compromised, use this API to res
 
 > **Caution:** Resetting signature secret will invalidate user's issued token! In other words if the user is already login, it will be logout from the system and can't make any further request until login again.
 
-> **Note:** Only an account admin can a user's secret!
+> **Note:** Only an account admin or the user himself can reset a user's secret!
```
Comment (jamesaimonetti) [inline applications/crossbar/doc/auth.md]: I think the sentence got clunky. What about "Only the user or an account admin can reset the user's secret"?

**Comment by danielfinke** [comment]
@jamesaimonetti ok, changed

**Comment by lazedo** [comment]
@danielfinke thanks for the update. i was thinking in extending `PUT` with action. can you check if its doable to add a function clause in `validate_action(Context, <<"reset_signature_secret">>, ?HTTP_PUT) ->` ?
thanks

**Comment by danielfinke** [comment]
@lazedo OK it is now part of the actions

File: applications/crossbar/src/modules/cb_auth.erl [hunk 2]
```diff
@@ -150,9 +151,14 @@ authorize(Context, PathToken, Id) ->
 
 %% authorize /auth
 -spec authorize_nouns(cb_context:context(), req_verb(), req_nouns()) -> boolean().
-authorize_nouns(C, ?HTTP_PUT, [{<<"auth">>, _}]) -> cb_context:is_superduper_admin(C);
+authorize_nouns(C, ?HTTP_PUT, [{<<"auth">>, _}]) -> authorize_action(C, cb_context:req_value(C, <<"action">>));
 authorize_nouns(C, ?HTTP_PUT, [{<<"auth">>, _}, {<<"accounts">>, _}]) -> cb_context:is_account_admin(C);
-authorize_nouns(C, ?HTTP_PUT, [{<<"auth">>, _}, {<<"users">>, _}, {<<"accounts">>, _}]) -> cb_context:is_account_admin(C);
+authorize_nouns(C, ?HTTP_PUT, [{<<"auth">>, _}, {<<"users">>, [UserId]}, {<<"accounts">>, [AccountId]}]) ->
```
Comment (lazedo) [inline applications/crossbar/src/modules/cb_auth.erl]: @danielfinke is this needed here ?

Comment (danielfinke) [reply applications/crossbar/src/modules/cb_auth.erl]: @lazedo yeah, the change here allows users to change their own signature secrets

**Comment by danielfinke** [comment]
@lazedo @jamesaimonetti anything else required for this PR?

**Comment by jamesaimonetti** [comment]
Yay, merged!

**Comment by danielfinke** [comment]
@jamesaimonetti could you do me a favour and resolve/close the JIRA ticket KAZOO-5697 I had created for this

**Pull request closed without merging**